### PR TITLE
DeviceConnector: Various changes

### PR DESCRIPTION
**Pull request by Gozzim**
## Changes Proposed:

- Secure connection support
- Complete error handling
- Log-Level support
- Huge CleanUp
- Various minor changes

### CODE REVIEW

File: packages/device-connector/src/deployment-server/connection.ts [hunk 1]
```diff
@@ -39,23 +41,24 @@ const readConnectorData = async (): Promise<ConnectorData> => {
...
-    }
+  if (!registrationResponse.ok) {
+    throw httpError(registrationResponse)
+  }
 
-    const connectorData: ConnectorData = {
-      id: id,
-      uri: address
-    }
+  const { id } = await registrationResponse.json() as any
```
Comment (WoH) [inline packages/device-connector/src/deployment-server/connection.ts:52]:
```suggestion
  const { id } = await registrationResponse.json() as {id: string}
```

File: packages/device-connector/src/deployment-server/connection.ts [hunk 2]
```diff
@@ -39,23 +41,24 @@ const readConnectorData = async (): Promise<ConnectorData> => {
...
 
-    resolve(connectorData)
-  })
+  if (id === '') {
+    throw new Error('No valid id response')
+  }
+
+  return {
+    id: id,
```
Comment (WoH) [inline packages/device-connector/src/deployment-server/connection.ts:59]:
```suggestion
    id,
```

File: packages/device-connector/src/deployment-server/connection.ts [hunk 3]
```diff
@@ -79,23 +82,36 @@ export const openStream = async (): Promise<Duplex> => {
...
-  if ((connectorData == null) || connectorData.uri !== address) {
-    connectorData = await generateConnectorData()
-    await writeConnectorData(connectorData)
+  if (connectorData == null || connectorData.id == null || connectorData.uri !== deployUri) {
+    logger.debug('No or invalid previous connection found for requested deployment-server')
+    try {
+      connectorData = await generateConnectorData()
+    } catch (e) {
+      logger.error(e)
+      await setTimeout(3000)
```
Comment (WoH) [inline packages/device-connector/src/deployment-server/connection.ts:94]: Might want to configure this via ENV in the future

File: packages/device-connector/src/device-types/store.ts [hunk 4]
```diff
@@ -6,12 +6,24 @@ import { logger } from '../util/logger'
 export const DeviceTypes = {
   store: new Array<DeviceType>(),
 
-  withFQBN (fqbn: FQBN): DeviceType | undefined {
-    return this.store.find((item) => item.fqbn === fqbn)
+  withFQBN (fqbn: FQBN): DeviceType {
```
Comment (WoH) [inline packages/device-connector/src/device-types/store.ts:9]: (find)byFqbn

would be my initial suggestion

File: packages/device-connector/src/device-types/store.ts [hunk 5]
```diff
@@ -6,12 +6,24 @@ import { logger } from '../util/logger'
...
+    if (devType == null) {
+      throw new Error(`No DeviceType with fqbn ${fqbn} found locally`)
+    }
+
+    return devType
   },
 
-  withId (typeId: string): DeviceType | undefined {
-    return this.store.find((item) => item.id === typeId)
+  withId (typeId: string): DeviceType {
```
Comment (WoH) [inline packages/device-connector/src/device-types/store.ts:19]: (find)byId, see above

File: packages/device-connector/src/devices/deployment.ts [hunk 6]
```diff
@@ -64,6 +64,7 @@ export const deployBinary = async (deployData: DeploymentData): Promise<void> =>
   }
 
   const artifactPath = await downloadArtifact(artifactUri)
+  logger.debug(`Deployment: ${deployData.id} - Downloaded artifact from ${artifactUri} to ${Path.basename(artifactPath)}`)
```
Comment (WoH) [inline packages/device-connector/src/devices/deployment.ts:67]:
```suggestion
  logger.debug(`Deployment: ${deployData.id} - Downloaded artifact from ${artifactUri} to ${basename(artifactPath)}`)
```

`import * as Path from 'path'` is an unconventional import

**Review by WoH — APPROVED** [review]
Small suggestions, no major concerns

File: packages/device-connector/src/deployment-server/connection.ts (hunk 1 above)
Comment (Gozzim) [reply packages/device-connector/src/deployment-server/connection.ts:52]: ah thanks alot. Was wondering how to set the type here

File: packages/device-connector/src/device-types/store.ts (hunk 4 above)
Comment (Gozzim) [reply packages/device-connector/src/device-types/store.ts]: refactored in 37f119b16a9a2ece04526ab67f65e60611370b7b

File: packages/device-connector/src/device-types/store.ts (hunk 5 above)
Comment (Gozzim) [reply packages/device-connector/src/device-types/store.ts]: refactored in 37f119b16a9a2ece04526ab67f65e60611370b7b

File: packages/device-connector/src/devices/deployment.ts (hunk 6 above)
Comment (Gozzim) [reply packages/device-connector/src/devices/deployment.ts]: changed in 37f119b16a9a2ece04526ab67f65e60611370b7b

File: packages/device-connector/src/deployment-server/connection.ts (hunk 3 above)
Comment (Gozzim) [reply packages/device-connector/src/deployment-server/connection.ts:94]: Possible change for future changes imo

**Pull request merged**